src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -68,6 +68,7 @@ class UDPPeerGroup[M](val config: Config)(implicit codec: Codec[M], bufferInstan
 
               val channel = activeChannels(channelId)
               messageE.foreach(message => channel.messageSubject.onNext(message))
+              datagram.content().release()
```

Comment:
should use try/finally for this too. onNext should not throw in theory but it could do in practice.